### PR TITLE
feat: Document sensitive info rule

### PR DESCRIPTION
**Pull request by e-moran**
This is the first draft of the docs.
I still want to do one more pass first though and add a bit more content about the custom detect function, but its like 90% of the way there.
I also need to remove shield from all of the examples. Will have this more polished tomorrow morning.

### CODE REVIEW

File: src/content/docs/sensitive-info/quick-start/bun.mdx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  </TabItem>
+</Tabs>
+
+### 4. Start app
+
+Start your app and load `http://localhost:3000`. Refresh the page and you will
+see the requests showing up in the [Arcjet dashboard](https://app.arcjet.com).
+Your entire application is protected.
+
+### 5. Simulate a suspicious request
```
Comment (davidmytton) [inline src/content/docs/sensitive-info/quick-start/bun.mdx:76]: This isn't the correct example

File: src/content/docs/sensitive-info/quick-start/nodejs.mdx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+Arcjet to block requests containing email addresses then try and send an email address.
+
+```sh
+curl -v http://localhost:8000 --data "My email address is test@example.com"
+```
+
+The default response for a request containing unexpected sensitive information is a 400:
+
+```sh
+# ...5 requests made previously
```
Comment (davidmytton) [inline src/content/docs/sensitive-info/quick-start/nodejs.mdx:120]: This comment refers to shield

File: src/content/docs/sensitive-info/quick-start/snippets/bun/Step3.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import arcjet, { sensitiveInfo } from "@arcjet/bun";
+import { env } from "bun";
+
+// This function is called by the `sensitiveInfo` rule to perform custom detection on strings.
```
Comment (davidmytton) [inline src/content/docs/sensitive-info/quick-start/snippets/bun/Step3.js:4]: Let's leave the custom detect function out of the quick start and move it to the reference. Quick start is about the bare minimum and then the reference can discuss the advanced options.

File: src/content/docs/sensitive-info/reference/snippets/nextjs/PerRouteApp.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import arcjet, { shield } from "@arcjet/next";
+import { NextResponse } from "next/server";
+
+const aj = arcjet({
+  key: process.env.ARCJET_KEY,
+  rules: [
+    shield({
```
Comment (davidmytton) [inline src/content/docs/sensitive-info/reference/snippets/nextjs/PerRouteApp.ts:7]: The sensitiveInfo rule is missing

File: src/content/docs/shield/quick-start/nextjs.mdx
Comment (davidmytton) [inline src/content/docs/shield/quick-start/nextjs.mdx:1]: Did you mean to edit this?

File: src/content/docs/shield/quick-start/sveltekit.mdx
Comment (davidmytton) [inline src/content/docs/shield/quick-start/sveltekit.mdx:1]: Did you mean to edit this?

File: src/content/docs/sensitive-info/quick-start/snippets/bun/Step3.js (hunk 3 above)
Comment (e-moran) [reply src/content/docs/sensitive-info/quick-start/snippets/bun/Step3.js:4]: I've added an example to the bun reference, if you're happy with it I'll copy it across to the other references.

Comment (davidmytton) [reply src/content/docs/sensitive-info/quick-start/snippets/bun/Step3.js:4]: Did you push it?

Comment (davidmytton) [reply src/content/docs/sensitive-info/quick-start/snippets/bun/Step3.js:4]: @e-moran I assume you did this in https://github.com/arcjet/arcjet-docs/pull/48/commits/d93ee1f8d03825dd7ce4537442d43c1c110c7636 ? The reference example looks good, but the quick start still has the custom detect.

File: src/content/docs/sensitive-info/quick-start/snippets/bun/Step3.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+const aj = arcjet({
+  key: env.ARCJET_KEY!, // Get your site key from https://app.arcjet.com
+  rules: [
+    // This allows all sensitive entities other than email addresses and those containing a dash character.
+    sensitiveInfo({
+      // allow: ["EMAIL"], Will block all sensitive information types other than email.
+      deny: ["EMAIL", "CONTAINS_DASH"], // Will block email and any custom detected values that "contain dash"
+      mode: "LIVE", // Will block requests, use "DRY_RUN" to log only
+      detect: detectDash,
+      contextWindowSize: 2, // Two tokens will be provided to the custom detect function at a time.
```
Comment (davidmytton) [inline src/content/docs/sensitive-info/quick-start/snippets/bun/Step3.ts:22]: This isn't explained anywhere - can you add a section in the reference to describe what it does and when you'd use it?

File: src/content/docs/sensitive-info/quick-start/bun.mdx (hunk 1 above)
Comment (davidmytton) [reply src/content/docs/sensitive-info/quick-start/bun.mdx:76]:
```suggestion
### 5. Test sending personal info
```

File: src/content/docs/sensitive-info/quick-start/nextjs.mdx [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+/>
+</TabItem>
+</Tabs>
+
+### 4. Start app
+
+Start your app and load `http://localhost:3000`. Refresh the page and you will
+see the requests showing up in the [Arcjet dashboard](https://app.arcjet.com).
+
+### 5. Simulate a suspicious request
```
Comment (davidmytton) [inline src/content/docs/sensitive-info/quick-start/nextjs.mdx:108]:
```suggestion
### 5. Test sending personal info
```

File: src/content/docs/sensitive-info/quick-start/nodejs.mdx [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+  ```
+  </TabItem>
+  <TabItem label="JS">
+  ```sh
+  node --env-file .env.local index.js
+  ```
+  </TabItem>
+</Tabs>
+
+### 5. Simulate a suspicious request
```
Comment (davidmytton) [inline src/content/docs/sensitive-info/quick-start/nodejs.mdx:107]:
```suggestion
### 5. Test sending personal info
```

File: src/content/docs/sensitive-info/quick-start/sveltekit.mdx [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+</TabItem>
+</Tabs>
+
+### 4. Start app
+
+Start your app and load `http://localhost:5173`. Refresh the page and you will
+see the requests showing up in the [Arcjet dashboard](https://app.arcjet.com).
+Your entire application is protected.
+
+### 5. Simulate a suspicious request
```
Comment (davidmytton) [inline src/content/docs/sensitive-info/quick-start/sveltekit.mdx:92]:
```suggestion
### 5. Test sending personal info 
```

File: src/content/docs/sensitive-info/quick-start/snippets/bun/Step3.ts (hunk 5 above)
Comment (davidmytton) [reply src/content/docs/sensitive-info/quick-start/snippets/bun/Step3.ts:22]: This = `contextWindowSize`

Comment (e-moran) [reply src/content/docs/sensitive-info/quick-start/snippets/bun/Step3.ts:22]: Yeah I'll add one in.

Comment (e-moran) [reply src/content/docs/sensitive-info/quick-start/snippets/bun/Step3.ts]: I've added this to the custom detect function section in the bun reference docs. Once you're happy with it I'll copy that section across to the other adapters.

File: src/content/docs/sensitive-info/quick-start/snippets/bun/Step3.js (hunk 3 above)
Comment (e-moran) [reply src/content/docs/sensitive-info/quick-start/snippets/bun/Step3.js]: That is sorted now.

**Review by davidmytton — COMMENTED** [review]
I made a few minor tweaks to the Bun quick start and reference, but this is ready to copy over to the other frameworks now.

**Comment by e-moran** [comment]
> I made a few minor tweaks to the Bun quick start and reference, but this is ready to copy over to the other frameworks now.

Copied that across

**Pull request merged**